examples/asa/assets/5-contract-owned-asa-approval.py

```diff
@@ -0,0 +1,80 @@
+from pyteal import *
+
+def contract_owned_asa():
+    '''
+    This program is stateful part of contract owned ASA, This program keeps track of
+    number of ASA created by the contract, It also checks for
+    : While payment only creator can receive the payments
+    : While asset creation only creator can create assets
+    Owner can also be changed using this program
+    '''
+
+    on_creation = Seq([
+        App.globalPut(Bytes("Creator"), Txn.sender()),
+        App.globalPut(Bytes("ASACounter"), Int(0)),
+        Return(Int(1))
+    ])
+
+    # verify neither transaction
+    # contains a rekey
+    rekey_check = And(
+        Gtxn[0].rekey_to() == Global.zero_address(),
+        Gtxn[1].rekey_to() == Global.zero_address()
+    )
+
+    on_asa_create = Seq([
+        Assert(And(
+            App.globalGet(Bytes("Creator")) == Txn.sender(),
+            App.globalGet(Bytes("ASACounter")) == Int(0), # limit no of asa from here: currently set to max 1 ASA.
```

Comment:
Shouldn't this be `App.globalGet(Bytes("ASACounter")) <= Int(1)` (If max 1 asa creation is allowed) ? If not then please explain this in comment

Reply:
No, because if you dry run this code, on first iteration it will create Asa therefore global will be 1 ( even if you add a counter + 1) , in next iteration it will still pass according to your logic, therefore we have created 2 ASAs now.

Reply:
Then can we do it this way:
```py
max_assets = Int(1)
...
on_asa_create = Seq([
  App.globalGet(Bytes("ASACounter")) == max_assets - Int(1) // explain here why we subtract 1 here (as ASA is already deployed)
  ..
])
```
If this way is possible then tt's easier to understand, and is more flexible (user can just update max_assets to Int(2) if (s)he wishes limit more than 1 ASA)